### ISSUE TITLE
Auto layout fails

### ISSUE THREAD

**Issue by ar1a** [geheur/bank-tag-custom-layouts#32]
Trying to generate an auto layout for a simple gear setup, fails with

```
2021-07-05 23:43:20 [Client] ERROR injected-client - Error in JavaScriptCallback
java.lang.IndexOutOfBoundsException: Index -1 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Unknown Source)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Unknown Source)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Unknown Source)
	at java.base/java.util.Objects.checkIndex(Unknown Source)
	at java.base/java.util.ArrayList.get(Unknown Source)
	at com.banktaglayouts.LayoutGenerator.basicLayout(LayoutGenerator.java:66)
	at com.banktaglayouts.LayoutGenerator.basicLayout(LayoutGenerator.java:21)
	at com.banktaglayouts.BankTagLayoutsPlugin.showLayoutPreview(BankTagLayoutsPlugin.java:398)
	at com.banktaglayouts.BankTagLayoutsPlugin.lambda$updateButton$1(BankTagLayoutsPlugin.java:193)
	at df.e(Varcs.java:149)
	at gk.f(WorldMapSection1.java:109)
	at ai.iv(MouseHandler.java:8826)
	at client.copy$menuAction(Client.java:8686)
	at hf.ir(GameObject.java:1457)
	at client.copy$menu(Client.java:7804)
	at client.id(Client.java:1856)
	at client.fr(Client.java:3606)
	at client.aw(Client.java:1295)
	at aa.s(GameEngine.java:336)
	at aa.run(GameEngine.java:315)
	at java.base/java.lang.Thread.run(Unknown Source)
2021-07-05 23:43:21 [Client] ERROR injected-client - Error in JavaScriptCallback
java.lang.NullPointerException: null
	at com.banktaglayouts.BankTagLayoutsPlugin.applyLayoutPreview(BankTagLayoutsPlugin.java:338)
	at com.banktaglayouts.BankTagLayoutsPlugin.lambda$updateButton$2(BankTagLayoutsPlugin.java:208)
	at df.e(Varcs.java:149)
	at gk.f(WorldMapSection1.java:109)
	at ai.iv(MouseHandler.java:8826)
	at client.copy$menuAction(Client.java:8686)
	at hf.ir(GameObject.java:1457)
	at client.copy$menu(Client.java:7869)
	at client.id(Client.java:1856)
	at client.fr(Client.java:3606)
	at client.aw(Client.java:1295)
	at aa.s(GameEngine.java:336)
	at aa.run(GameEngine.java:315)
	at java.base/java.lang.Thread.run(Unknown Source)
```

Associated line is https://github.com/geheur/bank-tag-custom-layouts/blob/badbf64b31433c3c4fa992d4a7faed7d4a1f4962/src/main/java/com/banktaglayouts/LayoutGenerator.java#L66

![image](https://user-images.githubusercontent.com/8436007/124496483-0d4b6f80-ddec-11eb-97cb-ced66614c13e.png) here are my settings

**Comment by ar1a** [comment]
Looked into this a bit more, it seems it fails if it wants to create more duplicates than the setting allows.

I have 5 super comb potions in my inventory, but the setting only allows 4. Once I set it the duplicates to 28, it works fine.

Same with turning create duplicates off

**Comment by geheur** [comment]
should be fixed in d1585256c1aced1349106422eb6b575afbd52046, ty for the bug report.
won't be on the phub immediately, but when it is it'll probably be v1.4.4

**Comment by ar1a** [comment]
thanks for the fast fix!
